library/common.c

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Internal invasive testing helper functions
```

Comment:
I'm not convinced that a `common.c` is the right place for `mbedtls_err_add`.

Given that this is a test hook, we could live with having it in `error.c` and only enabled when error messages are enabled.

In any case, `common.c` is not an acceptable name for the sake of Mbed OS (Mbed TLS is periodically imported into the Mbed OS source tree). The Mbed OS build system has this peculiarity that it doesn't accept two `.c` files with the same name, even in different directories belonging to different subprojects. So we shouldn't use names that are too generic for `library/*.c`. Anything that's obviously crypto/TLS related is fine (e.g. `aes.c`, `ssl_*.c`, `x509_*.c`). We got away with `debug.c` but it's not something we would have created after Mbed OS became relevant. Let's not add `common.c` even if nobody has nabbed it yet (I haven't checked).

Reply:
I [originally](https://github.com/ARMmbed/mbedtls/pull/4006/commits/f6b76e3ba8326dd40c4870dc884a6fdf087356a9) had placed `mbedtls_err_add` in `error.c` however I moved it when I realised that `error.c` was an auto-generated file. I will look into adding it through the script that generates it.

Reply:
Done.